core/input/input.cpp

```diff
@@ -1164,6 +1166,29 @@ void Input::_button_event(int p_device, JoyButton p_index, bool p_pressed) {
 }
 
 void Input::_axis_event(int p_device, JoyAxis p_axis, float p_value) {
+	int axis_id = (int)p_axis;
```

Comment:
Does removing cast cause any warning?

Reply:
It is even an error when compiling on Linux:
```
./core/input/input.cpp:1203:6: error: cannot initialize a variable of type 'int' with an lvalue of type 'JoyAxis'
        int axis_id = p_axis;
```